src/Core/Framework/Api/HealthCheck/Model/Result.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Shopware\Core\Framework\Api\HealthCheck\Model;
+
+use InvalidArgumentException;
+
+class Result
+{
+    public function __construct(
+        private readonly string $name,
+        private readonly Status $status = Status::Healthy,
+        private readonly string $errorMessage = ''
+    )
+    {
+    }
+
+    public function name(): string
```

Comment:
Maybe we can just make these all public props instead, possibly keeping `healthy` ?